### PR TITLE
[world-vercel] Handle null key response from run-key API endpoint

### PR DESCRIPTION
**Pull request by TooTallNate**
## Summary

- Update `fetchRunKey` to handle the new `{ key: null }` response from the Vercel API `GET /v1/workflow/run-key/:deploymentId` endpoint, returning `undefined` to signal encryption is disabled for that workflow run.
- Update the Zod response schema to accept `z.string().nullable()` and the return type to `Promise<Uint8Array | undefined>`.
- Add tests for `fetchRunKey` covering the null key, valid key, and error response cases.

## Context

The Vercel API endpoint now returns `{ key: null }` instead of a 404 when a deployment does not have an encryption key. Without this change, `fetchRunKey` would throw a Zod validation error on the `null` value.

All callers already handle `undefined` as "no encryption" via the existing `rawKey ? await importKey(rawKey) : undefined` pattern.

### CODE REVIEW

File: packages/world-vercel/src/encryption.ts [hunk 1]
```diff
@@ -122,10 +123,13 @@ export async function fetchRunKey(
   }
 
   const data = await response.json();
-  const result = z.object({ key: z.string() }).safeParse(data);
+  const result = z.object({ key: z.string().nullable() }).safeParse(data);
   if (!result.success) {
     throw new Error('Invalid response from Vercel API, missing "key" field');
```
[the remark this answers is not in the document]
Comment (TooTallNate) [reply packages/world-vercel/src/encryption.ts]: Good catch — updated the error message to include the expected shape and the Zod error details for easier debugging.

File: packages/world-vercel/src/encryption.test.ts [hunk 2]
```diff
@@ -84,3 +84,53 @@ describe('deriveRunKey', () => {
...
+        new Response(JSON.stringify({ key: null }), { status: 200 })
+      );
+
+    const result = await fetchRunKey(deploymentId, testProjectId, testRunId, {
+      token: 'test-token',
+    });
+
+    expect(result).toBeUndefined();
+    fetchSpy.mockRestore();
+  });
```
[the remark this answers is not in the document]
Comment (TooTallNate) [reply packages/world-vercel/src/encryption.test.ts:105]: Agreed — moved mock restoration to an `afterEach` with `vi.restoreAllMocks()` so cleanup always runs even if an assertion throws.

File: packages/world-vercel/src/encryption.test.ts [hunk 3]
```diff
@@ -84,3 +84,53 @@ describe('deriveRunKey', () => {
...
+      );
+
+    const result = await fetchRunKey(deploymentId, testProjectId, testRunId, {
+      token: 'test-token',
+    });
+
+    expect(result).toBeInstanceOf(Uint8Array);
+    expect(result).toEqual(Buffer.from(keyBase64, 'base64'));
+    fetchSpy.mockRestore();
+  });
```
[the remark this answers is not in the document]
Comment (TooTallNate) [reply packages/world-vercel/src/encryption.test.ts:119]: Fixed — same `afterEach` cleanup covers this test as well.

File: packages/world-vercel/src/encryption.test.ts [hunk 4]
```diff
@@ -84,3 +84,53 @@ describe('deriveRunKey', () => {
...
+      .mockResolvedValueOnce(new Response('Not found', { status: 404 }));
+
+    await expect(
+      fetchRunKey(deploymentId, testProjectId, testRunId, {
+        token: 'test-token',
+      })
+    ).rejects.toThrow('HTTP 404');
+
+    fetchSpy.mockRestore();
+  });
```
[the remark this answers is not in the document]
Comment (TooTallNate) [reply packages/world-vercel/src/encryption.test.ts:131]: Fixed — same `afterEach` cleanup covers this test as well.
